web_server/posts/views.py

```diff
@@ -1,135 +1,86 @@
+            comments_json = [comment['fields'] for comment in comments_json]
+            json_posts[i]['fields']['comments'] = comments_json
+
+
+        # And filter out the meta data from the top level object
+        for post in json_posts:
+            post['fields']['id'] = post['pk']
+        json_posts = [post['fields'] for post in json_posts]
 
         output = {
+            "query": "posts",
             "count": pager.count,
-            "size": size,
-            # @todo how to implement better serialization of posts in a general way?
-            "posts": [
-                {
-                    "author": post.author.display_name,
-                    "content": post.content
-                } for post in pager.get_page(page)
-            ]
+            "size": len(posts),
+            "posts": json_posts
         }
-
-        pagination_uri = req.build_absolute_uri()
-        # Page and size might not have been passed in to this request, so we need to build a pagination url
-        # that includes these variables. However we cannot simply modify the request object, so we must
-        # deconstruct, modify, and reconstruct the uri.
-
-        # START STACKOVERFLOW DERIVATIVE CONTENT
-        # This solution has been provided via StackOverflow, and the following code snippet is CC-BY-SA
-        # Original Question: https://stackoverflow.com/questions/5755150/altering-one-query-parameter-in-a-url-django
-        # Question By: EvdB https://stackoverflow.com/users/5349/evdb
-        # Answer By: Tom Christie https://stackoverflow.com/users/596689/tom-christie
-        (scheme, netloc, path, params, query, fragment) = urlparse(pagination_uri)
-
-        next_query_dict = QueryDict(query).copy()
-        # We only set the next page if we are not on the last page
-        if page < pager.num_pages:
-            next_query_dict['page'] = int(next_query_dict['page']) + 1 if page in next_query_dict else 2
-            next_query = next_query_dict.urlencode()
-            output['next'] = urlunparse((scheme, netloc, path, params, next_query, fragment))
-
-        prev_query_dict = QueryDict(query).copy()
-        # We only set the previous page if we are not on the first page
-        if page > 1:
-            prev_query_dict['page'] = int(prev_query_dict['page']) - 1  # if page in prev_query_dict else 1
-            prev_query = prev_query_dict.urlencode()
-            output['prev'] = urlunparse((scheme, netloc, path, params, prev_query, fragment))
-        # END STACKOVERFLOW DERIVATIVE CONTENT
+        (prev_uri, next_uri) = pagination_uris
+        if prev_uri:
+            output['prev'] = prev_uri
+        if next_uri:
+            output['next'] = next_uri
 
         return JsonResponse(output)
 
-    def html_handler(req):
-        # @todo add pagination behaviour once it can be extracted from other handler
-        posts = Post.objects.filter(visibility="PUBLIC")
+    def html_handler(posts, pager, pagination_uris):
```

Comment:
I am a bit confused about the purpose of the html handler. The API only requires we return with application/json, so why include an html handler?

Reply:
The API is for other servers to talk to our server, but the endpoint location is also the URI for users on our server to view the posts